tests/generation/test_utils.py

```diff
@@ -453,13 +481,12 @@ def test_greedy_generate_dict_outputs(self):
                 # Retrocompatibility check
                 self.assertIsInstance(output_generate, GreedySearchDecoderOnlyOutput)
 
-            self._check_outputs(output_generate, main_input, model.config)
+            self._check_outputs(output_generate, model.config)
```

Comment:
We actually don't need `main_input`, we can obtain the batch size directly from the many outputs of `generate`